### PR TITLE
Add a user-agent prefix by default.

### LINKED ISSUE #191: Automatically set the user-agent string on each request.
The client should set the user agent prefix in each request.  Call the `SetUserAgentPrefix()` member function in the `grpc::ChannelArguments` before creating any channels in `bigtable::DataClient` or `bigtable::AdminClient`.  The prefix should include the library version, e.g.,

```C++
auto prefix = absl::StrCat("cbt-cpp/", bigtable::major_version(), ".", bigtable::minor_version(), "-dev");
```

The "-dev" suffix should be (manually) removed when we create real release candidates.  We should consider appending the output of 

```commandline
$ git rev-parse --short HEAD
```

to the user agent in dev versions.  The output from `git rev-parse` that can be captured by cmake into
a generated `.cc` file.

### PR DESCRIPTION
**Pull request by coryan**
This fixes #191.  Every grpc::Channel created by the client
library includes the library name (`cbt-c++`) and version as
a user-agent prefix.  The version includes the git commit (the
short form of the SHA-1), which is captured by CMake at
build-time.  I also captured the compiler id, compiler version,
and relevant compiler flags because those will be useful
elsewhere.

There is some overlap with #195, whichever one gets merged first I will resolve the conflicts.

### CODE REVIEW

File: bigtable/client/version.h [hunk 1]
```diff
@@ -15,6 +15,8 @@
 #ifndef GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_VERSION_H_
 #define GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_VERSION_H_
 
+#include <string>
+
 #ifndef _MSC_VER
 // Microsoft Vistual Studio does not define __cplusplus correctly for C++11.
```
Comment (mbrukman) [inline bigtable/client/version.h]: s/Vistual/Visual/

File: bigtable/client/client_options.cc [hunk 2]
```diff
@@ -27,6 +27,7 @@ ClientOptions::ClientOptions() {
     credentials_ = grpc::GoogleDefaultCredentials();
   }
   channel_arguments_ = grpc::ChannelArguments();
+  channel_arguments_.SetUserAgentPrefix("cbt-c++/" + version_string());
```
Comment (mbrukman) [inline bigtable/client/client_options.cc]: `StrCat`?

File: bigtable/client/build_info.cc.in [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+// limitations under the License.
+
+#include "bigtable/client/build_info.h"
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+
+char const compiler[] = R"""(@CMAKE_CXX_COMPILER_ID@ @CMAKE_CXX_COMPILER_VERSION@)""";
+
+char const compiler_flags[] = R"""(@CMAKE_CXX_FLAGS@ ${CMAKE_CXX_FLAGS_${BUILD_TYPE_UPPER}})""";
```
Comment (mbrukman) [inline bigtable/client/build_info.cc.in]: This is going to get long, no? And it leaks local paths for, e.g., `-I` header paths, etc.? Are you sure you want to include that?

Comment (coryan) [reply bigtable/client/build_info.cc.in]: I think this is Okay.   `CMAKE_CXX_FLAGS` is only `-Wall -Werror`, and `CMAKE_CXX_FLAGS_${BUILD_TYPE_UPPER}` is something like `-g` or `-O3 -ONDEBUG`.

The include paths and such do not go into these variables, you can put them there, but that is bad form, and our CMake files do not do that.

Comment (mbrukman) [reply bigtable/client/build_info.cc.in]: SGTM, this works for me.

**Comment by mbrukman** [comment]
Also, please note that the newly-added AppVeyor build failed for this PR.

File: bigtable/client/client_options.cc (hunk 2 above)
Comment (coryan) [reply bigtable/client/client_options.cc]: Done.

File: bigtable/client/version.h (hunk 1 above)
Comment (coryan) [reply bigtable/client/version.h]: Fixed.

**Comment by coryan** [comment]
We should buy @paprat a cold beverage.  First PR after he completes the AppVeyor build and it catches a problem.  Yay!

**Pull request merged**